### DISCUSSION TITLE
function compilation error

### DISCUSSION THREAD

**Discussion by khateraAlizada** [FLAMEGPU/FLAMEGPU2#1020] [Q&A] [answered]
Define message section code: 
  message = model.newMessageBruteForce("ns_location_message")
  message.newVariableID("id")
  message.newVariableFloat("x")
  message.newVariableFloat("y")
  message.newVariableInt("has-i")
message.newVariableID("adopter_id") 

Define agent section code: 
    fn = ns.newRTCFunction("ns_adoptI3", ns_adoptI3)
    fn.setMessageInput("ns_location_message")
 my function: 
![Screen Shot 2022-12-15 at 7 46 43 AM](https://user-images.githubusercontent.com/20228364/207863444-da1cef56-f6d8-44db-a91e-fb4d49643844.png)
I get some error when I run the simulation : 
![Screen Shot 2022-12-15 at 8 17 48 AM](https://user-images.githubusercontent.com/20228364/207869035-ade0c81f-464a-480b-a6c8-086474670dcb.png)

**Comment by ptheywood** [comment, marked as answer]
Unfortunately Google Colab's version of `ipykernel` is very old and doesn't forward messages from `std::cout` to the user, so the actual compilation error message isn't visible when using google colab: see https://github.com/FLAMEGPU/FLAMEGPU2-tutorial-python/issues/10 and the issue we opened with googlecollab https://github.com/googlecolab/colabtools/issues/2230 requesting it be updated.

We are reliant on google collab making this change, it is not something we can fix ourselves.

---

Looking at the code block in the first screenshot, I think line 41 is your issue, as C/C++/CUDA variable names cannot contain `-`.

Using `const int has_i = ...` should resolve that problem at least.

**Reply by khateraAlizada** [reply]
Thank you very much for the response. I can see the logs:
Runtime ->view runtime logs

issue resolved.
